packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseSchemaObject-test.js

```diff
@@ -533,4 +533,202 @@ describe('Schema Object', () => {
       expect(element.attributes.get('typeAttributes').toValue()).to.deep.equal(['fixedType', 'nullable']);
     });
   });
+
+  describe('#description', () => {
+    it('warns when description is not a string', () => {
+      const schema = new namespace.elements.Object({
+        description: true,
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(2);
+
+      expect(parseResult).to.contain.warning(
+        "'Schema Object' 'description' is not a string"
+      );
+    });
+
+    it('adds a description to the returned element', () => {
+      const schema = new namespace.elements.Object({
+        description: 'an example schema',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.description.toValue()).to.equal('an example schema');
+    });
+  });
+
+  describe('#default', () => {
+    it('adds a default value to the returned element', () => {
+      const schema = new namespace.elements.Object({
+        default: 'my default',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('default').toValue()).to.equal('my default');
+    });
+
+    it('adds a default value to the returned element when matching type', () => {
+      const schema = new namespace.elements.Object({
+        type: 'string',
+        default: 'my default',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('default').toValue()).to.equal('my default');
+    });
+
+    it('adds a default value to the returned element when matching enum', () => {
+      const schema = new namespace.elements.Object({
+        enum: ['my default'],
+        default: 'my default',
+      });
+      const parseResult = parse(context, schema);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
+
+      const element = parseResult.get(0).content;
+      expect(element.attributes.get('default').toValue()).to.equal('my default');
+    });
+
+    it('allows a null default value when nullable is enabled', () => {
```

Comment:
I'd test null and enum interplay as well, it had some problems in the past IIRC

Reply:
The test already uses an enum with nullable, perhaps I mis-understand the feedback can you please check if the schema below is what you desire.

Reply:
You are right 🤦‍♂️ I was reading the test description.

So the test actually does what I wanted you to test, but the description doesn't indicate it, so enum is just an implementation detail of the test and not an explicit subject of the test. Someone could come in the future and change the test so it doesn't contain enum, but still tests a null default value when nullable is enabled.

For that reason I'd either do two tests, one with enum and one without, where the enum one explicitly mentions it i the test description, or just edit the description here to be explicit about testing the enum/nullable interplay as well.

Does it make sense what I'm writing?

Reply:
Certainly does, will add additional tests. There are actually three cases to test:

- when there is no type or enum
- there is a type
- there is an enum